### PR TITLE
Rewrite and document core DFT functionality

### PR DESCRIPTION
**Pull request by prehner**

### CODE REVIEW

**Comment by prehner** [comment]
grafik

This also adds the toc structure for the FeOs theory guide.

File: docs/theory/dft/euler_lagrange_equation.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+$$F_{\rho_i}^\mathrm{b}-\mu_i=0$$
+
+which can be inserted into (1) to give
+
+$$F_{\rho_i}(r)=F_{\rho_i}^\mathrm{b}-V_i^\mathrm{ext}(r)\tag{2}$$
+
+### Spherical molecules
+In the simplest case, the molecules under considerations can be described as spherical. Then the Helmholtz energy can be split in to an ideal and a residual part:
+
+$$\beta F=\sum_i\int\rho_i(r)\left(\ln\left(\rho_i(r)\Lambda_i^3\right)-1\right)dr+\beta F^\mathrm{res}$$
```
Comment (g-bauer) [inline docs/theory/dft/euler_lagrange_equation.md:27]: Maybe use square brackets?

`$$\beta F=\sum_i\int\rho_i(r)\left[\ln\left(\rho_i(r)\Lambda_i^3\right)-1\right]dr+\beta F^\mathrm{res}$$`

Comment (g-bauer) [inline docs/theory/dft/euler_lagrange_equation.md:27]: maybe add: `with the de Broglie wavelength $\Lambda_i$.`

File: docs/theory/dft/euler_lagrange_equation.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+$$\rho_i(r)=\rho_i^\mathrm{b}e^{\beta\left(F_{\rho_i}^\mathrm{b,res}-F_{\rho_i}^\mathrm{res}(r)-V_i^\mathrm{ext}(r)\right)}$$
+
+which is the common form of the Euler-Lagrange equation for spherical molecules.
+
+### Homosegmented chains
+For chain molecules that do not resolve individual segments (essentially the PC-SAFT Helmholtz energy functional) a chain contribution is introduced as
+
+$$\beta F^\mathrm{chain}=-\sum_i\int\rho_i(r)\left(m_i-1\right)\ln\left(\frac{y_{ii}\lambda_i(r)}{\rho_i(r)}\right)dr$$
+
+Here, $m_i$ is the chain length, $y_{ii}$ the cavity correlation function at contact in the reference fluid, and $\lambda_i$ a weighted density.
```
Comment (g-bauer) [inline docs/theory/dft/euler_lagrange_equation.md:46]: $m_i$ is the number of segments?

File: docs/theory/dft/euler_lagrange_equation.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+The presence of $\rho(r)$ in the logarithm poses numerical problems. Therefore, it is convenient to rearrange the expression as
+
+$$\beta F^\mathrm{chain}=\sum_i\int\rho_i(r)\left(m_i-1\right)\left(\ln\left(\rho_i(r)\Lambda_i^3\right)-1\right)dr\underbrace{-\sum_i\int\rho_i(r)\left(m_i-1\right)\left(\ln\left(y_{ii}\lambda_i(r)\Lambda_i^3\right)-1\right)dr}_{\beta\hat{F}^\mathrm{chain}}$$
```
Comment (g-bauer) [inline docs/theory/dft/euler_lagrange_equation.md:49]: Maybe use `\begin{align}` (see my comment at the end of the review) and break this equation into two lines?

proposal:
```latex
\begin{align}
\beta F^\mathrm{chain}=&\sum_i\int\rho_i(r)\left(m_i-1\right)\left[\ln\left(\rho_i(r)\Lambda_i^3\right)-1\right]dr \\
&\underbrace{-\sum_i\int\rho_i(r)\left(m_i-1\right)\left[\ln\left(y_{ii}\lambda_i(r)\Lambda_i^3\right)-1\right]dr}_{\beta\hat{F}^\mathrm{chain}}
\end{align}
```

File: docs/theory/dft/euler_lagrange_equation.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+Then the total Helmholtz energy
+
+$$\beta F=\sum_i\int\rho_i(r)\left(\ln\left(\rho_i(r)\Lambda_i^3\right)-1\right)dr+\beta F^\mathrm{chain}+\beta F^\mathrm{res}$$
```
Comment (g-bauer) [inline docs/theory/dft/euler_lagrange_equation.md:53]: If square brackets above:
`$$\beta F=\sum_i\int\rho_i(r)\left[\ln\left(\rho_i(r)\Lambda_i^3\right)-1\right]dr+\beta F^\mathrm{chain}+\beta F^\mathrm{res}$$`

File: docs/theory/dft/euler_lagrange_equation.md [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+can be rearranged to
+
+$$\beta F=\sum_i\int\rho_i(r)m_i\left(\ln\left(\rho_i(r)\Lambda_i^3\right)-1\right)dr+\underbrace{\beta\hat{F}^\mathrm{chain}+\beta F^\mathrm{res}}_{\beta\hat{F}^\mathrm{res}}$$
```
Comment (g-bauer) [inline docs/theory/dft/euler_lagrange_equation.md:57]: If square brackets above:
`$$\beta F=\sum_i\int\rho_i(r)m_i\left[\ln\left(\rho_i(r)\Lambda_i^3\right)-1\right]dr+\underbrace{\beta\hat{F}^\mathrm{chain}+\beta F^\mathrm{res}}_{\beta\hat{F}^\mathrm{res}}$$`

File: docs/theory/dft/euler_lagrange_equation.md [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+$$\beta F_{\rho_i}=m_i\ln\left(\rho_i(r)\Lambda_i^3\right)+\beta\hat{F}_{\rho_i}^\mathrm{res}$$
+
+$$\beta F_{\rho_i}^\mathrm{b}=m_i\ln\left(\rho_i^\mathrm{b}\Lambda_i^3\right)+\beta\hat{F}_{\rho_i}^\mathrm{b,res}$$
+
+and lead to a slightly modified Euler-Lagrange equation
+
+$$\rho_i(r)=\rho_i^\mathrm{b}e^{\frac{\beta}{m_i}\left(\hat F_{\rho_i}^\mathrm{b,res}-\hat F_{\rho_i}^\mathrm{res}(r)-V_i^\mathrm{ext}(r)\right)}$$
+
+### Heterosegmented chains
+Thex expressions are more complex for models in which density profiles of individual segments are considered. A derivation is given in the appendix of ***. The resulting Euler-Lagrange equation is given as
```
Comment (g-bauer) [inline docs/theory/dft/euler_lagrange_equation.md:70]: - The**~~x~~**
- `appendix of ***` missing link?

File: docs/theory/dft/functional_derivatives.md [hunk 7]
```diff
@@ -0,0 +1,39 @@
+## Functional derivatives
+
+In the last section the functional derivative
+
+$$\hat F_{\rho_\alpha}^\mathrm{res}(r)=\left(\frac{\delta F^\mathrm{res}}{\delta\rho_\alpha(r)}\right)_{T,\rho_{\alpha'\neq\alpha}}$$
```
Comment (g-bauer) [inline docs/theory/dft/functional_derivatives.md:5]: Missing `\hat` in derivative?

`$$\hat F_{\rho_\alpha}^\mathrm{res}(r)=\left(\frac{\delta \hat F^\mathrm{res}}{\delta\rho_\alpha(r)}\right)_{T,\rho_{\alpha'\neq\alpha}}$$`

File: docs/theory/dft/functional_derivatives.md [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+
+$$n_\gamma(r)=\sum_\alpha\int\rho_\alpha(r')\omega_\gamma^\alpha(r-r')dr'\tag{1}$$
+
+In practice the weight functions tend to have simple shapes like step functions (i.e. the weighted density is an average over a sphere) or Dirac distributions (i.e. the weighted density is an average over the surface of a sphere).
+
+For Helmholtz energy functionals that can be written in this form, the calculation of the functional derivative can be automated. In general the functional derivative can be written as
+
+$$F_{\rho_\alpha}(r)=\int\frac{\delta f(r')}{\delta\rho_\alpha(r)}dr'=\int\sum_\gamma f_{n_\gamma}(r')\frac{\delta n_\gamma(r')}{\delta\rho_\alpha(r)}dr'$$
+
+with $f_{n_\gamma}$ as abbreviation for the partial derivative $\frac{\partial f}{\partial n_\gamma}$. Using the definition of the weighted densities (1), the expression can be rewritten as
```
Comment (g-bauer) [inline docs/theory/dft/functional_derivatives.md:21]: maybe emphasize "partial"?

`...as abbreviation for the *partial* derivative...`

**Review by g-bauer — CHANGES_REQUESTED** [review]
We should add `amsmath` to the MyST-parser configuration (in `conf.py`) which allows to use [direct LaTeX Math](https://myst-parser.readthedocs.io/en/latest/syntax/optional.html#syntax-amsmath). This allows for `align` and `equation` blocks (and other stuff) that we can use to introduce line breaks in long formulas.

Nitpick: use upright `\mathrm{d}r` in integrals? No strong feelings about this, though.

File: docs/index.md [hunk 9]
```diff
@@ -131,4 +131,13 @@ recipes/index
 
 rustguide/index
 rust_api
+```
+
+```{toctree}
```
[the remark this answers is not in the document]
Comment (prehner) [reply docs/index.md:136]: I like the order Python -> Rust -> theory which basically goes from the surface to the core. Plus (for now) the theory guide is less complete than the other parts.

File: docs/theory/dft/euler_lagrange_equation.md (hunk 1 above)
Comment (prehner) [reply docs/theory/dft/euler_lagrange_equation.md:27]: In this context square brackets are already used to specify a functional dependency.

**Review by g-bauer — APPROVED** [review]

**Pull request merged**